Sunbird CDN configuration

surabhi-mahawar opened the pull request:
**Jira ticket:** [SB-31158](https://project-sunbird.atlassian.net/browse/SB-31158)

**Description:** This PR includes the environment variables required for sunbird cloud agnostic tool. We have added conditions to use this tool or other services based on the configuration. This tool provides option to use AWS/Azure etc which will also be handled via sunbird confirguation variables **SUNBIRD_CLOUD_MEDIA_STORAGE_TYPE**.

We intend to store all the media files(Image/Audio/Video/Document) being used in UCI assessment bot. Also use this to display media files as well.

If these configuration is not set, it will start the service but will just show error message in the service logs and will not respond to that particular media related message.

surabhi-mahawar commented:
@keshavprasadms 
Can you please merge this PR.

keshavprasadms commented:
@surabhi-mahawar 
- Please add a details description on what this change is about
- Jira link references
- Documentation link if any
- What do you intend to store in these buckets / containers
- What would happen if CDN / media storage is not configured for an environment
- What about other CSP's implementation

surabhi-mahawar commented:
@keshavprasadms 
I have updated the description of the PR with jira ticket and the required information you mentioned. 
We have added these environment variables for the sunbird's cloud agnostic tool only.

keshavprasadms commented on ansible/roles/stack-sunbird/templates/transformer.env:
```diff
@@ -72,6 +72,14 @@ POSTHOG_EVENT_ENABLED=FALSE
-AZURE_BLOB_STORE_ACCOUNT_NAME={{sunbird_private_storage_account_key}}
-SELECTED_FILE_CDN=azure
\ No newline at end of file
+AZURE_BLOB_STORE_ACCOUNT_NAME={{sunbird_private_storage_account_name}}
```
@surabhi-mahawar Are these variable still required? Below is another variable which has same contents
`SUNBIRD_CLOUD_MEDIA_STORAGE_KEY`

surabhi-mahawar replied:
No this is not required not, I have removed these variables from the files

keshavprasadms commented:
@surabhi-mahawar This PR is supporting only Azure cloud right? In the description, I can see AWS support also, but I am unable to find the corresponding AWS cloud variables in the PR.

surabhi-mahawar commented:
> @surabhi-mahawar This PR is supporting only Azure cloud right? In the description, I can see AWS support also, but I am unable to find the corresponding AWS cloud variables in the PR.

@keshavprasadms The sunbird cloud agnostic tool should handle this according to the **SUNBIRD_CLOUD_MEDIA_STORAGE_TYPE** mentioned, if this changes to AWS, the credentials for it should be added in **SUNBIRD_CLOUD_MEDIA_STORAGE_KEY, SUNBIRD_CLOUD_MEDIA_STORAGE_SECRET** as well. 
It should change for everyone right, not just for one instance. 
It should be either Azure or AWS. 

Or can there be a case where different instances can have different CDNs ?

keshavprasadms commented:
@surabhi-mahawar Please can you connect with @santhosh-tg for this? He is driving a change across all BB's to standardize the variable names and the naming convention, how to support multiple clouds etc. A brief discussion with him would help you get clarity and take this PR forward. Would be good if you can please setup a call and discuss with the team - @santhosh-tg @gandham-santhosh

surabhi-mahawar commented:
> @surabhi-mahawar Please can you connect with @santhosh-tg for this? He is driving a change across all BB's to standardize the variable names and the naming convention, how to support multiple clouds etc. A brief discussion with him would help you get clarity and take this PR forward. Would be good if you can please setup a call and discuss with the team - @santhosh-tg @gandham-santhosh

@keshavprasadms 
Can you please send us the email IDs for @santhosh-tg @gandham-santhosh, so we can connect with them.

keshavprasadms commented:
I don't think adding email id's here is a good idea due to privacy concerns. As @ChakshuGautam is already on Slack, you can easily get in touch with the folks. These are the slack handlers
- `@Santhosh Gandham`
- `@Santhosh Kumar Sreedharan`

surabhi-mahawar commented:
@keshavprasadms 
I have made the changes in the PR as per the variables mentioned in the sheet & discussion with Santhosh Kumar Sreedharan.
Can you please review this PR.

CC: @ChakshuGautam

santhosh-tg commented on ansible/roles/stack-sunbird/templates/inbound.env:
```diff
@@ -53,14 +53,16 @@ REDIS_PASS={{sunbird_redis_pass | default('')}}
 
+#Sunbird CDN Configuration
+SUNBIRD_CLOUD_MEDIA_STORAGE_TYPE={{cloud_service_provider}}
+SUNBIRD_CLOUD_MEDIA_STORAGE_KEY={{cloud_public_storage_accountname}}
```
@surabhi-mahawar  I think the storage account is private. Please check and confirm. If yes, change the accountname and secret vars accordingly.

surabhi-mahawar replied:
@santhosh-tg 
I have changed the storage account variables to private. 
Please review it.
